lib/statement-builder/statements/table.js

```diff
@@ -1,211 +1,203 @@
+    };
 
-  //
-  // If we are altering the table with a `with`
-  //
-  if (alter && Object.keys(alter).length) {
-    w = new With(alter);
-    if (w.error) return w.error;
-    opts.with = w.cql;
-  }
+    const tableName = this._computeTable(options);
 
-  //
-  // `ensure` or `drop` currently
-  //
-  opts.type = options.type;
+    this.params = [];
+
+    this.cql = this._compile(options, tableName, options.lookupKey || schema.primaryKeys());
 
-  var env = process.env.NODE_ENV;
-  if (['prod', 'production'].indexOf(env) !== -1
-      && opts.type === 'drop'
-      && !options.force) {
-    return new Error('Please don\'t try and drop your prod tables without being certain');
+    return this;
   }
 
-  return opts;
-};
+  _computeTable(options) {
+    const schema = options.schema || this.schema;
+    const type = options.type;
 
-TableStatement.prototype.build = function (options) {
-  var schema = options.schema || this.schema;
-  var type = options.type;
-  var tableName;
+    this.options.queryName = [type, 'index', schema.name, options.lookupKey].join('-');
 
-  this.options = {
-    executeAsPrepared: true,
-    queryName: type + '-table-' + schema.name
-  };
+    if (!options.lookupKey) return this.table;
 
-  tableName = this._computeTable(options);
+    const table = this.schema.lookupTables[options.lookupKey];
 
-  this.params = [];
+    if (table) return table;
+    //
+    // Compute the lookupTable name based on the key and if its used as an index
+    // or not
+    //
+    return options.useIndex
+      ? schema.name + '_' + options.lookupKey
+      : schema.name + '_by_' + options.lookupKey.replace(/_\w+$/, '');
+  }
 
-  this.cql = this._compile(options, tableName, options.lookupKey || schema.primaryKeys());
+  //
+  // Figure out what statement will be executed
+  //
+  _compile(options) {
+    const fn = this['_' + options.type];
+    if (fn) return fn.apply(this, arguments);
 
-  return this;
-};
+    // This shouldn't happen
+    throw new Error('Invalid type ' + options.type);
+  }
 
-TableStatement.prototype._computeTable = function (options) {
-  var schema = options.schema || this.schema;
-  var type = options.type;
-  var table;
+  /*
+  * Drop the table or index
+  * @options {Object} options passed in
+  * @tableName {String} Name of table or index
+  * @returns {String} cql value for statement to be executed
+  */
+  _drop(options, tableName) {
+    return [
+      'DROP',
+      (options.useIndex ? 'INDEX' : 'TABLE'),
+      tableName
+    ].join(' ');
+  }
 
-  this.options.queryName = [type, 'index', schema.name, options.lookupKey].join('-');
+  _ensure(options, tableName, primaryKeys) {
+    const schema = options.schema || this.schema;
+    const secondaryKeys = schema.secondaryKeys();
 
-  if (!options.lookupKey) return this.table;
+    tableName = tableName || this.table;
+    primaryKeys = primaryKeys || schema.primaryKeys();
 
-  table = this.schema.lookupTables[options.lookupKey];
+    let cql = '';
```

Comment:
This file specifically could use some template string love

Reply:
@scommisso: done